CMakeLists.txt

```diff
@@ -1,21 +1,31 @@
 cmake_minimum_required(VERSION 3.2)
-project(fzf C)
+project(fzf LANGUAGES C)
 
 add_library(${PROJECT_NAME} SHARED "src/fzf.c")
 
+set(is-windows $<STREQUAL:${CMAKE_SYSTEM_NAME},Windows>)
+
+target_include_directories(${PROJECT_NAME} PUBLIC
+  $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src>)
+target_compile_options(${PROJECT_NAME}
+  PRIVATE
+    $<$<C_COMPILER_ID:MSVC>:/W4>
+    $<$<NOT:$<C_COMPILER_ID:MSVC>>:-Wall>)
```

Comment:
I would probably make separate configuration per build type btw, see https://gitlab.kitware.com/cmake/cmake/-/issues/20812

Reply:
https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#genex:C_COMPILER_ID

These are generator expressions, not variables.

Reply:
Additionally, per-config build types do not make much sense when setting warning flags. The issue here is that Clang on Windows when using the MSVC ABI (the default LLVM install does this) does not know what `/W4` is. We're simply switching off whatever default C++ compiler someone has that CMake finds.

Reply:
> These are generator expressions, not variables.

Thanks kitware! 😄 

> Additionally, per-config build types do not make much sense when setting warning flags.

Not sure I follow, what I basically meant it that we should not force _additional_ warning flags for a release build, instead, we could have a debug/dev build which includes whatever flags are needed for testing.

Reply:
> Not sure I follow, what I basically meant it that we should not force additional warning flags for a release build

Ahh, I wasn't trying to argue in favor of adding flags. These warning flags are already present in the current `CMakeLists.txt`. I was trying to keep parity with current behavior, but offload behavior to generation time instead of configuration time. 🙂